internal/xds/v3/contour.go

```diff
@@ -151,7 +151,10 @@ func (s *contourServer) stream(st grpcStream) error {
 
 			// Rewrite the embedded message types to v3.
 			for _, r := range resources {
-				xds.Rewrite(r)
+				// Note that the resource cache
+				// gave us a pointer to its internal
+				// object, so we should rewrite a copy.
+				xds.Rewrite(proto.Clone(r))
```

Comment:
don't we need to *do* something with the result of this rewrite? E.g. collect them and iterate over them in L160? Seems like as-is we're rewriting and throwing away the rewritten result.